tests/Kernel/fixtures/patterns/date_block_pattern_rendering.yml

```diff
@@ -0,0 +1,91 @@
+# Test "default" variant.
+- variant: default
+  date:
+    start: 370173600
+    end: NULL
+    timezone: Europe/Brussels
+  assertions:
+    count:
+      div.ecl-date-block.ecl-date-block--default: 1
+      div.ecl-date-block.ecl-date-block--cancelled: 0
+      div.ecl-date-block.ecl-date-block--past: 0
+      div.ecl-date-block.ecl-date-block--ongoing: 0
+
+    equals:
+      span.ecl-date-block__week-day: Thu
+      span.ecl-date-block__day: 24
+      span.ecl-date-block__month: Sep
+      span.ecl-date-block__year: 1981
+
+# Test "cancelled" variant.
+- variant: cancelled
+  date:
+    start: 370173600
+    end: NULL
+    timezone: Europe/Brussels
+  assertions:
+    count:
+      div.ecl-date-block.ecl-date-block--default: 0
+      div.ecl-date-block.ecl-date-block--cancelled: 1
+      div.ecl-date-block.ecl-date-block--past: 0
+      div.ecl-date-block.ecl-date-block--ongoing: 0
+
+    equals:
+      span.ecl-date-block__week-day: Thu
+      span.ecl-date-block__day: 24
+      span.ecl-date-block__month: Sep
+      span.ecl-date-block__year: 1981
+
+# Test "ongoing" variant.
+- variant: ongoing
+  date:
+    start: 370173600
+    end: NULL
+    timezone: Europe/Brussels
+  assertions:
+    count:
+      div.ecl-date-block.ecl-date-block--default: 0
+      div.ecl-date-block.ecl-date-block--cancelled: 0
+      div.ecl-date-block.ecl-date-block--past: 0
+      div.ecl-date-block.ecl-date-block--ongoing: 1
+    equals:
+      span.ecl-date-block__week-day: Thu
+      span.ecl-date-block__day: 24
+      span.ecl-date-block__month: Sep
+      span.ecl-date-block__year: 1981
+
+# Test "past" variant.
+- variant: past
+  date:
+    start: 370173600
+    end: NULL
+    timezone: Europe/Brussels
+  assertions:
+    count:
+      div.ecl-date-block.ecl-date-block--default: 0
+      div.ecl-date-block.ecl-date-block--cancelled: 0
+      div.ecl-date-block.ecl-date-block--past: 1
+      div.ecl-date-block.ecl-date-block--ongoing: 0
+    equals:
+      span.ecl-date-block__week-day: Thu
+      span.ecl-date-block__day: 24
+      span.ecl-date-block__month: Sep
+      span.ecl-date-block__year: 1981
+
+# Test "unknown" variant.
+- variant: unexistant
```

Comment:
This variant is out of scope and I don't see the purpose.
This comment is not a blocker.

Reply:
I think this should be part of the test, to check if when rendering a variant that is unknown, it doesn't set any specific existing classes.

Reply:
I think it's good to test what happen when the variant is either empty or not a valid one, we should then fallback to the default variant. We can add this logic directly in the pattern template or in the global pattern preprocess.

Reply:
Fixed!